demo/src/index.ts

```diff
@@ -0,0 +1,120 @@
+import { Point } from '@noble/secp256k1'
+import delay from 'delay'
+import { BigNumber, utils, Wallet } from 'ethers'
+import { existsSync } from 'fs'
+import { readFile, writeFile } from 'fs/promises'
+import { hideBin } from 'yargs/helpers'
+import yargs from 'yargs/yargs'
+import { ProofRequest } from '@e2e-zk-ecdsa/shared'
+
+import { URLS } from './constants'
+
+const fetchErc20AnonSet = async ({
+  min,
+  tokenAddress,
+}: {
+  tokenAddress: string
+  min: string
+}) => {
+  const params = new URLSearchParams({
+    min,
+    tokenAddress,
+  })
+
+  const res = await fetch(
+    `${URLS.QUERY_API}/balance/ERC20?${params.toString()}`,
+  )
+  return res.json()
+}
+
+const createProofRequest = async ({
+  addresses,
+  privateKey,
+}: {
+  addresses: string[]
+  privateKey: string
+}) => {
+  const wallet = new Wallet(privateKey)
+  const { address, publicKey } = wallet
+  const point = Point.fromHex(publicKey.slice(2))
+  const signature = await wallet.signMessage(1234n.toString())
+
+  return new ProofRequest({
+    addresses: addresses.map((address) => BigNumber.from(address).toBigInt()),
+    addressIndex: addresses.indexOf(address),
+    msghash: 1234n,
+    pubkey: point,
+    signature: utils.arrayify(signature),
+  })
+}
+const fetchProof = async ({
+  addresses,
+  privateKey,
+}: {
+  addresses: string[]
+  privateKey: string
+}) => {
+  console.log('Creating proof request')
+  const proofRequest = await createProofRequest({
+    addresses,
+    privateKey,
+  })
+
+  await delay(300)
+  const body = proofRequest.stringify()
+  console.log({ proofRequest: body })
+  await delay(300)
+  console.log('Sending proof request to proving API')
+
+  const response = await fetch(URLS.PROVING_API, {
```

Comment:
This is timing out with `HeadersTimeoutError: Headers Timeout Error` from `undici`. (This is really the client fetch request timing out, not the prover server).

@BlakeMScurr Was this why, in your code to test the client you used [http.request](https://github.com/privacy-scaling-explorations/e2e-zk-ecdsa/blob/95b388975f4fcd95afd57410c7b46bdddc420c0a/apis/proving/test/test_client.ts#L61)?

Reply:
Hmm, nope I wasn't that thoughtful about my choice of `http.request`. It just seems like a solution, and it seemed to work.

Reply:
Yes it does work, because unlike node.fetch it doesn'ttime out